src/LiveComponent/tests/Integration/LiveComponentHydratorTest.php

```diff
@@ -144,6 +144,140 @@ public function testCanDehydrateAndHydrateComponentWithTestCases(callable $testF
+                ->assertObjectAfterHydration(function (object $object) {
+                    $this->assertSame('Revert to Chicken', $object->product->name);
+                });
+        }];
+
+        yield 'onUpdated: with IDENTITY' => [function () {
+            $entityOriginal = create(Entity1::class)->object();
+            $entityNext = create(Entity1::class)->object();
+            \assert($entityOriginal instanceof Entity1);
+            \assert($entityNext instanceof Entity1);
+
+            return HydrationTest::create(new class() {
+                #[LiveProp(writable: [LiveProp::IDENTITY], onUpdated: [LiveProp::IDENTITY => 'onEntireEntityUpdated'])]
+                public Entity1 $entity1;
+
+                public function onEntireEntityUpdated($oldValue)
+                {
+                    // Sanity check
+                    if ($this->entity1 === $oldValue) {
+                        throw new \Exception('Old value is the same entity?!');
+                    }
+                    if (2 === $this->entity1->id) {
+                        // Revert the value
+                        $this->entity1 = $oldValue;
+                    }
+                }
+            })
+                ->mountWith(['entity1' => $entityOriginal])
+                ->userUpdatesProps(['entity1' => $entityNext->id])
+                ->assertObjectAfterHydration(function (object $object) use ($entityOriginal) {
+                    $this->assertSame(
+                        $entityOriginal->id,
+                        $object->entity1->id
+                    );
+                });
+        }];
+
+        yield 'onUpdated: with IDENTITY to null' => [function () {
+            $entity1 = create(Entity1::class)->object();
+            \assert($entity1 instanceof Entity1);
+
+            return HydrationTest::create(new class() {
+                #[LiveProp(writable: [LiveProp::IDENTITY], onUpdated: [LiveProp::IDENTITY => 'onEntireEntityUpdated'])]
+                public ?Entity1 $entity1;
+
+                public function onEntireEntityUpdated($oldValue)
+                {
+                    // Sanity check
+                    if ($this->entity1 === $oldValue) {
+                        throw new \Exception('Old value is the same entity?!');
+                    }
+                    if (null === $this->entity1) {
+                        // Revert the value
+                        $this->entity1 = $oldValue;
+                    }
+                }
+            })
+                ->mountWith(['entity1' => $entity1])
+                ->userUpdatesProps(['entity1' => null])
+                ->assertObjectAfterHydration(function (object $object) use ($entity1) {
+                    $this->assertNotNull($object->entity1);
+                    $this->assertSame(
+                        $entity1->id,
+                        $object->entity1->id
+                    );
+                });
+        }];
```

Comment:
Especially once you do a little refactoring, I think this test won't be needed either (just keep the one basic test for the IDENTITY above)